sdk/eventhubs/azure_messaging_eventhubs/tests/round_trip.rs

```diff
@@ -0,0 +1,169 @@
+    common::setup();
+    let host = env::var("EVENTHUBS_HOST").unwrap();
+    let eventhub = env::var("EVENTHUB_NAME").unwrap();
+    let producer = ProducerClient::new(
+        host.clone(),
+        eventhub.clone(),
+        DefaultAzureCredential::create(TokenCredentialOptions::default()).unwrap(),
+        ProducerClientOptions::builder()
+            .with_application_id("test_new")
+            .build(),
+    );
+
+    assert!(producer.open().await.is_ok());
+
+    let partition_properties = producer
+        .get_partition_properties(EVENTHUB_PARTITION)
+        .await
+        .unwrap();
+
+    info!(
+        "Start receiving messages from sequence: {:?}",
+        partition_properties.last_enqueued_sequence_number
+    );
+
+    let start_sequence = partition_properties.last_enqueued_sequence_number;
+    let mut batch = producer
+        .create_batch(Some(
+            EventDataBatchOptions::builder()
+                .with_partition_key("My Partition Key.")
+                .with_partition_id(EVENTHUB_PARTITION)
+                .build(),
+        ))
+        .await
+        .unwrap();
+
+    assert!(batch
+        .try_add_event_data(
+            EventData::builder()
+                .with_body(b"Hello, World!")
+                .add_property("Message#", 1)
+                .with_message_id(1)
+                .build(),
+            None
+        )
+        .unwrap());
+
+    assert!(batch
+        .try_add_amqp_message(
+            AmqpMessage::builder()
+                .with_body(AmqpValue::from("Hello, World!"))
+                .add_application_property("Message#", 2)
+                .with_properties(AmqpMessageProperties::builder().with_message_id(2).build())
+                .build(),
+            None,
+        )
+        .unwrap());
+
+    assert!(batch
+        .try_add_amqp_message(
+            AmqpMessage::builder()
+                .with_body(AmqpList::from(vec![
+                    AmqpValue::from("Hello, World!"),
+                    3.into(),
+                    5.into()
+                ]))
+                .add_application_property("Message#", 3)
+                .with_properties(AmqpMessageProperties::builder().with_message_id(3).build())
+                .build(),
+            None,
+        )
+        .unwrap());
+
+    assert!(batch
+        .try_add_amqp_message(
+            AmqpMessage::builder()
+                .add_application_property("Message#", 4)
+                .with_properties(AmqpMessageProperties::builder().with_message_id(4).build())
+                .build(),
+            None
+        )
+        .unwrap());
+
+    assert!(producer.submit_batch(&batch).await.is_ok());
+
+    let consumer = ConsumerClient::new(
+        host,
+        eventhub,
+        None,
+        DefaultAzureCredential::create(TokenCredentialOptions::default()).unwrap(),
+        Some(
+            ConsumerClientOptions::builder()
+                .with_application_id("test_new")
```

Comment:
Share the application Id in a variable so that both producer and consumer change the value in one place

Reply:
Makes sense.